client/app/bundles/course/courses/components/tables/PendingTodosTable.tsx

```diff
@@ -0,0 +1,281 @@
+
+interface Props extends WrappedComponentProps {
+  todos: TodoData[];
+  todoType: 'assessments' | 'videos' | 'surveys';
+}
+
+const translations = defineMessages({
+  pendingAssessmentsHeader: {
+    id: 'course.courses.show.pendingAssessments',
+    defaultMessage: 'Pending Assessments',
+  },
+  pendingVideosHeader: {
+    id: 'course.courses.show.pendingVideos',
+    defaultMessage: 'Pending Videos',
+  },
+  pendingSurveysHeader: {
+    id: 'course.courses.show.pendingSurveys',
+    defaultMessage: 'Pending Surveys',
+  },
+  seeMoreFailure: {
+    id: 'course.courses.show.seeMoreFailure',
+    defaultMessage: 'Failed to load more pending tasks',
+  },
+  tableHeaderTitle: {
+    id: 'course.courses.show.tableHeaderTitle',
+    defaultMessage: 'Title',
+  },
+  tableHeaderStartAt: {
+    id: 'course.courses.show.tableHeaderStartAt',
+    defaultMessage: 'Start At',
+  },
+  tableHeaderEndAt: {
+    id: 'course.courses.show.tableHeaderEndAt',
+    defaultMessage: 'End At',
+  },
+  tableSeeMore: {
+    id: 'course.courses.show.tableSeeMore',
+    defaultMessage: 'SEE MORE',
+  },
+  accessButtonRespond: {
+    id: 'course.courses.show.accessButtonRespond',
+    defaultMessage: 'Respond',
+  },
+  accessButtonEnterPassword: {
+    id: 'course.courses.show.accessButtonEnterPassword',
+    defaultMessage: 'Enter Password',
+  },
+  accessButtonAttempt: {
+    id: 'course.courses.show.accessButtonAttempt',
+    defaultMessage: 'Attempt',
+  },
+  accessButtonResume: {
+    id: 'course.courses.show.accessButtonResume',
+    defaultMessage: 'Resume',
+  },
+  accessButtonWatch: {
+    id: 'course.courses.show.accessButtonWatch',
+    defaultMessage: 'Watch',
+  },
+});
+
+const PendingTodosTable: FC<Props> = (props) => {
+  const { intl, todos, todoType } = props;
+
+  const [shavedTodos, setShavedTodos] = useState(todos.slice(0, 5));
+  const [end, setEnd] = useState(10);
+
+  useEffect(() => {
+    setShavedTodos(todos.slice(0, end - 5));
+  }, [todos]);
+
+  const renderButtons = (todo: TodoData): JSX.Element => {
+    let accessButtonText = '';
+    let accessButtonLink = '';
+    switch (todoType) {
+      case 'surveys':
+        accessButtonText = intl.formatMessage(translations.accessButtonRespond);
+        if (todo.progress === 'in_progress') {
+          accessButtonLink = getSurveyResponseURL(
+            getCourseId(),
+            todo.itemActableId,
+            todo.itemActableSpecificId,
+          );
+        } else {
+          accessButtonLink = getSurveyURL(getCourseId(), todo.itemActableId);
+        }
+        break;
+
+      case 'assessments':
+        if (todo.progress === 'not_started') {
+          if (!todo.canAccess && todo.canAttempt) {
+            accessButtonText = intl.formatMessage(
+              translations.accessButtonEnterPassword,
+            );
+            accessButtonLink = getAssessmentURL(
+              getCourseId(),
+              todo.itemActableId,
+            );
+          } else {
```

Comment:
not elsif todo.canAttempt ?

![image](https://user-images.githubusercontent.com/42570513/176091699-1eff1f6e-3db8-409a-aa15-c89ae65faf9b.png)

seems like it should be a post request, not get.